lib/matplotlib/figure.py

```diff
@@ -1759,6 +1759,25 @@ def get_tightbbox(self, renderer=None, bbox_extra_artists=None):
 
         return _bbox
 
+    @staticmethod
+    def _norm_per_subplot_kw(per_subplot_kw):
+        expanded = {}
+        for k, v in per_subplot_kw.items():
+            if isinstance(k, tuple):
+                for sub_key in k:
+                    if sub_key in expanded:
+                        raise ValueError(
+                            f'The key {sub_key!r} appears multiple times.'
```

Comment:
Should that necessarily be an error? We could also define sequential application to override more general settings (in a sense we already to that with `per_subplot_kw > subplot_kw`).

That would allow
```
{
    ('A', 'B', 'C', 'D'): {'key1': val1, 'key2': val2},
    'D': {'key3', val3},
}
```
instead of having to split and repeat
```
{
    ('A', 'B', 'C'): {'key1': val1, 'key2': val2},
    'D': {'key1': val1, 'key2': val2, 'key3', val3},
}
```

But I can also see a point that this may be more confusing/error prone and we should stick to a more narrow definition. Probably also depends how many different kws we expect.

Reply:
I did think about this a bit, but was not sure what the precedence rule should be.  dicts are ordered but that feels a bit brittle to depend on and checking that they did not overlap seemed like more complexity than it was worth.

If we get issues and change our mind about this we can lossen up without too much worry about back API